File: leetcode/0238-product-of-array-except-self/solution.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> left(n);
        vector<int> right(n);
        exclusive_scan(nums.begin(), nums.end(),
                        left.begin(),
                        1, multiplies<int>());
        exclusive_scan(nums.rbegin(), nums.rend(),
                        right.rbegin(),
                        1, multiplies<int>());
        transform (left.begin(), left.end(), 
            right.begin(), right.begin(), 
            multiplies<int>());
        return right;
    }
};

File: leetcode/0238-product-of-array-except-self/solution.go
func productExceptSelf(nums []int) []int {
    n := len(nums);
    left := make([]int, n)
    left[0] = 1;
    right := make([]int, n)
    right[n-1] = 1;
    for i := 1; i<n;i++ {
        left[i] = left[i-1]*nums[i-1]
    }
    for i := n-2; i>=0;i-- {
        right[i] = right[i+1]*nums[i+1]
    }
    ret := make([]int, n)
    for i := 0; i<n;i++ {
        ret[i] = left[i]*right[i]
    }
    return ret
}
